cpp/src/arrow/csv/parser.cc

```diff
@@ -324,9 +324,29 @@ class BlockParserImpl {
         if (*(end - 1) == '\r') {
           --end;
         }
-        return MismatchingColumns(batch_.num_cols_, num_cols,
-                                  first_row_ < 0 ? -1 : first_row_ + batch_.num_rows_,
-                                  util::string_view(start, end - start));
+        int32_t batch_row = batch_.num_rows_ + batch_.num_skipped_rows();
```

Comment:
`num_skipped_rows` is potentially O(n)... so, say you have one skipped row every two rows, parsing becomes O(n²)?

Reply:
Done